Direction finding connectionless locator sample app: add support for nRF5340

ppryga opened the pull request:
Update configuration to allow build of the sample application with nRF5340.
Add information about nRF5340 support in README.rst.

This PR requires following other PRs to be merged:

- https://github.com/zephyrproject-rtos/zephyr/pull/36047 to be merged in upstream and cherry-picked into sdk-zephyr repository.
- https://github.com/nrfconnect/sdk-nrf/pull/4763 to be merged into main.
- Upmerge of sdk-zephyr repository has to be finished.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4773
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf:
```diff
@@ -0,0 +1,20 @@
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_SYNC_PERIODIC=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
```
Remove duplicated options:

CONFIG_BT_CTLR_ADV_EXT=y
CONFIG_BT_CTLR_SYNC_PERIODIC=y

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf:
```diff
@@ -0,0 +1,27 @@
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_SW_SPLIT=y
+
+CONFIG_BT_CTLR_ADV_EXT=y
```
Remove duplicated options:
CONFIG_BT_CTLR_ADV_EXT=y
CONFIG_BT_CTLR_SYNC_PERIODIC=y

ppryga replied:
Removed

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/README.rst:
```diff
@@ -16,7 +16,7 @@ The sample supports the following development kits:
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
-   :rows: nrf52833dk_nrf52833
+   :rows: nrf52833dk_nrf52833, nrf5340dk_nrf5340_cpuapp_and_cpuappns
```
Can we use `nrf5340dk_nrf5340_cpuapp_and_cpuappns`, or should we split this into 2 boards?

ppryga replied:
`nrf5340dk_nrf5340_cpuapp_and_cpuappns` is used by other BLE samples in nrf-sdk, so I assume that this it the correct one.

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/README.rst:
```diff
@@ -38,6 +38,15 @@ Configuration
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
+
+Board specific configuration involves configuring the Bluetooth LE controller configuration.
```
`Board specific configuration involves configuring the Bluetooth LE controller`

ppryga replied:
Removed 'controller' word.

mia-ko replied:
Why remove the 'controller' word?
I think this sentence should be: Board-specific configuration involves configuring the Bluetooth LE controller.
Right?

ppryga replied:
@mia-ko I've put wrong word in comment. The removed word is ` configuration`.

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.conf:
```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Reguired to enable BT_BUF_CMD_TX_SIZE for LE Set Extended Advertising Data command
```
Required

ppryga replied:
Corrected, thank you.

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay:
```diff
@@ -0,0 +1,28 @@
+	 */
+	dfe-antenna-num = < 12 >;
+	/* This is a setting that enables antenna 12 (in antenna matrix designed
+	 * by Nordic) for Tx PDU. For more information see README.rst.
```
RX ?

ppryga replied:
Corrected, thank you.

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/child_image/hci_rpmsg.overlay:
```diff
@@ -0,0 +1,28 @@
+
+	/* These are GPIO pin numbers that are provided to
+	 * Radio peripheral. The pins will be acquired by Radio to
+	 * drive antenna switching when AoD is enabled.
```
AoA ?

ppryga replied:
Corrected, thank you.

MarekPieta commented on samples/bluetooth/direction_finding_connectionless_rx/README.rst:
```diff
@@ -46,6 +55,8 @@ To build this sample with angle of departure mode only, set ``OVERLAY_CONFIG`` t
 See :ref:`cmake_options` for instructions on how to add this option.
 For more information about using configuration overlay files, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
 
+To build this sample for :ref:`nRF5340 DK <ug_nrf5340>`, with angle of arrival mode only, add content of :file:`overlay-aoad.conf` file to :file:`child_image/hci_rpmsg.conf` file.
```
`overlay-aod`

ppryga replied:
Corrected, thank you.

mia-ko commented on samples/bluetooth/direction_finding_connectionless_rx/README.rst:
```diff
@@ -38,6 +38,15 @@ Configuration
+This sample configuration is split into the following two files:
+
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
```
```suggestion
* board specific configuration is available in :file:`boards/<BOARD>.conf` file
```

ppryga replied:
Corrected, thank you.

mia-ko reviewed (changes requested):
Just one typo left! Otherwise it looks good :)

ppryga commented:
PR converted to Draft on request from @thst-nordic due to limited CI resources. Reviewing and approval process still continues without change.

KAGA164 commented on samples/bluetooth/direction_finding_connectionless_rx/boards/nrf52833dk_nrf52833.conf:
```diff
@@ -0,0 +1,18 @@
+
+# Enable Direction Finding Feature including AoA and AoD
+CONFIG_BT_CTLR_DF=y
+CONFIG_BT_CTLR_DF_ANT_SWITCH_TX=n
```
Shouldn't that be under line 17 for clarity ?

ppryga replied:
Moved below line `# Enable Direction Finding Feature including AoA and AoD`

ppryga commented:
Moved DTS overlay from `child_image/hci_rpmsg.overlay` to `child_image/hci_rpmsg/nrf5340dk_nrf5340_cpunet.overlay`.
Corrected antenna switches GPIO mapping in the DTS overlay.

ppryga commented:
@KAGA164 I've provided requested change. Take a look on PR, please.

mbolivar-nordic reviewed (changes requested):
> This PR requires following other PRs to be merged:
> [...]
> [nrf fromtree] Add support for nRF5340 in direction finding sdk-zephyr#578 to be merget into sdk-zephyr.
> [nrf fromtree] Add support for nRF52820 in direction finding sdk-zephyr#577 to be merget into sdk-zephyr.

Please consolidate the two sdk-zephyr PRs into a single PR, and pull them into this PR's west.yml, so we can get CI results.

mbolivar-nordic reviewed (approved):
Changes look good, thanks for bringing in the manifest update.

ppryga commented:
Rebased again due to issues in CI jobs.

mbolivar-nordic commented:
This will have to wait to be merged until after the upmerge I just started.

ppryga commented:
Removed commit that updated west manifest.
